src/lib.rs

```diff
@@ -34,17 +34,17 @@ pub trait AsStrFormatExt: AsRef<str> {
     /// assert_eq!("{}a{}b{}c".format(&[1, 2, 3, 4]), "1a2b3c"); // extra arguments are ignored
     /// assert_eq!("{}a{}b{}c".format(&[1, 2]), "1a2bc"); // missing arguments are replaced by empty string
     /// assert_eq!("{{}}{}".format(&[1, 2]), "{}1");
-    fn format<'a, T: Display + ?Sized + 'a>(&self, args: impl IntoIterator<Item=&'a T> + Clone) -> String {
+    fn format<'a, T: Display>(&self, args: &[T]) -> String {
```

Comment:
[T] is too rigid. Can you use `Index` trait?

Reply:
I tried it, but the problem is that the args[i] panics if i >= args.len() .
I think this is hardly suitable in our case.
I use args.get(i) and if the index >= args.len()  - just take the next argument with args.next().
As far as I understand, there is no non-panicking alternative to Index trait in rust for now ..

Reply:
Ok